fiftyone/utils/labels.py

```diff
@@ -1198,3 +1206,223 @@ def _get_filepath(sample_or_frame, sample):
+            dict which contains the keys required by the camera_model for each
+            sample in the sample_collection.
+        forward_transform_flags: a dict mapping `transformation_key_field` to
+            lists of booleans indicating whether to apply forward or inverse
+            transformations for each transform in the list of transformations
+            for each sample
+        camera_model (fou3d.pinhole_projector): a callable that takes 3D points
+            in the camera coordinate system and the camera parameters dict and
+            returns the projected 2D points in pixel coordinates
+        transformation_key_field ("id"): the key field to use for looking up
+            transformations for each sample
+        camera_key_field ("id"): the key field to use for looking up camera
+            parameters for each camera sample
+        batch_size (1000): number of samples to process in each batch
+        progress (None): whether to render a progress bar (True/False), use the
+            default value ``fiftyone.config.show_progress_bars`` (None), or a
+            progress callback function to invoke instead
+    """
+    fov.validate_grouped_non_dynamic_collection(sample_collection)
+    fov.validate_collection_label_fields(
+        sample_collection, in_field, fol.Detections
+    )
+
+    camera_slice = sample_collection.select_group_slices(camera_slice_name)
+    camera_slice.compute_metadata(progress=progress)
+
+    values_2d_label = {}
+    count = 0
+
+    for group in sample_collection.select_fields(in_field).iter_groups(
+        group_slices=[spatial_slice_name, camera_slice_name],
+        progress=progress,
+        autosave=True,
+    ):
+        spatial_sample = group[spatial_slice_name]
+        camera_sample = group[camera_slice_name]
+        spatial_key = spatial_sample[transformation_key_field]
+        camera_key = camera_sample[camera_key_field]
+        transforms = transformations.get(spatial_key, None)
+        forward_flags = (
+            forward_transform_flags.get(spatial_key)
+            if forward_transform_flags
+            else None
+        )
+        cam_params = camera_params.get(camera_key, None)
+
+        if transforms is None:
+            logger.warning(
+                f"Skipping sample {spatial_sample.id} because transformations are missing"
+            )
+            continue
+        if cam_params is None:
+            logger.warning(
+                f"Skipping camera sample {camera_sample.id} because camera parameters are missing"
+            )
+            continue
+
+        width, height = (
+            camera_slice[camera_sample.id].metadata.width,
+            camera_slice[camera_sample.id].metadata.height,
+        )
+
+        if width is None or height is None:
+            logger.warning(
+                f"Skipping camera sample {camera_sample.id} because camera metadata is missing. Please compute metadata for this sample first."
```

Comment:
you already called compute metadata https://github.com/voxel51/fiftyone/pull/6277/files#diff-41b9587fdc5ac2683dcc9b89656cf2664c7a38aa9ec55b35f02cc032e87e961dR1270 so if it doesn't have metadata, there is probably an error